consensus/XDPoS/engines/engine_v2/engine.go

```diff
@@ -393,7 +393,7 @@ func (x *XDPoS_v2) Finalize(chain consensus.ChainReader, header *types.Header, s
 		if len(common.StoreRewardFolder) > 0 {
 			data, err := json.Marshal(rewards)
 			if err == nil {
-				err = os.WriteFile(filepath.Join(common.StoreRewardFolder, header.Number.String()+"."+header.Hash().Hex()), data, 0644)
+				err = os.WriteFile(filepath.Join(common.StoreRewardFolder, header.Number.String()), data, 0644)
```

Comment:
why remove hash from file name ?

Reply:
the hash in the filename doesn't match the hash from XDPoS_getV2BlockByNumber_committed because after finalize, there are further changes to block contents

`header.Root = state.IntermediateRoot(chain.Config().IsEIP158(header.Number))
	header.UncleHash = types.CalcUncleHash(nil)`

in Seal()
`header.Validator = signature`